drivers/sensor/ti/ina23x/ina230.c

```diff
@@ -20,21 +18,53 @@ LOG_MODULE_REGISTER(INA230, CONFIG_SENSOR_LOG_LEVEL);
 
 /** @brief The LSB value for the bus voltage register, in microvolts/LSB. */
 #define INA230_BUS_VOLTAGE_UV_LSB 1250U
+#define INA236_BUS_VOLTAGE_UV_LSB 1600U
 
 /** @brief The scaling for the power register. */
 #define INA230_POWER_SCALING 25
+#define INA236_POWER_SCALING 32
+
+/** @brief ADCRANGE bit in configuration register */
+#define INA236_CONFIG_ADC_RANGE BIT(12)
+
+enum ina23x_ids {
+	ina230,
+	ina236
+};
+
+struct ina23x_config {
+	uint16_t bus_voltage_uv_lsb;
+	uint8_t power_scaling;
+	bool have_high_precision;
+};
+
+const static struct ina23x_config ina23x_config[] = {
+	/* clang-format off */
+	[ina230] = {
+			.bus_voltage_uv_lsb = INA230_BUS_VOLTAGE_UV_LSB,
+			.power_scaling = INA230_POWER_SCALING,
+			.have_high_precision = false
+		},
+	[ina236] = {
+			.bus_voltage_uv_lsb = INA236_BUS_VOLTAGE_UV_LSB,
+			.power_scaling = INA236_POWER_SCALING,
+			.have_high_precision = true
```

Comment:
Trailing commas would format it better, in my opinion, but it is not compliant with the code style, and the pre-push hook will reject it with this error `ERROR:OPEN_BRACE: that open brace { should be on the previous line`.

There are these options now:

1. Leave it as it is
2. Add a trailing comma after the last element of the `ina23x_config` array
3. Add a trailing comma into each initialization of an element in the array
4. Add trailing commas after the last element and into each element initialization (will be rejected)
5. Create a macro to initialize the structure
6. Do not use `ina23x_config` at all and pass the configuration in the `DT_INST_FOREACH_STATUS_OKAY_VARGS`

If the second suggestion is applied it will look like this (looks good and compliant):

```c
const static struct ina23x_config ina23x_config[] = {
        [ina230] = {.bus_voltage_uv_lsb = INA230_BUS_VOLTAGE_UV_LSB,
                    .power_scaling = INA230_POWER_SCALING,
                    .have_high_precision = false},
        [ina236] = {.bus_voltage_uv_lsb = INA236_BUS_VOLTAGE_UV_LSB,
                    .power_scaling = INA236_POWER_SCALING,
                    .have_high_precision = true},
};
```

If the third option is applied (elements are misaligned):

```c
const static struct ina23x_config ina23x_config[] = {
        [ina230] =
                { 
                        .bus_voltage_uv_lsb = INA230_BUS_VOLTAGE_UV_LSB,
                        .power_scaling = INA230_POWER_SCALING,
                        .have_high_precision = false,
                },
        [ina236] = {
                .bus_voltage_uv_lsb = INA236_BUS_VOLTAGE_UV_LSB,
                .power_scaling = INA236_POWER_SCALING,
                .have_high_precision = true,
        }};
```

The macro version could be something like this:

```c
#define CREATE_CONFIG(LSB, SCALING, HIGH_PRECISION)                                                \
        {                                                                                          \
                .bus_voltage_uv_lsb = LSB, .power_scaling = SCALING,                               \
                .have_high_precision = HIGH_PRECISION,                                             \
        }

const static struct ina23x_config ina23x_config[] = {
        [ina230] = CREATE_CONFIG(INA230_BUS_VOLTAGE_UV_LSB, INA230_POWER_SCALING, false),
        [ina236] = CREATE_CONFIG(INA236_BUS_VOLTAGE_UV_LSB, INA236_POWER_SCALING, true),
};
```

If you can think of some other way, please suggest it.

Reply:
I like the idea of a dedicated macro. If a new variant were to be added, it would likely only add one extra line into `ina23x_config`.

Reply:
I updated it with the macro version